### PR TITLE
Improve rendering of the tabs 

### LINKED ISSUE #298: [FEATURE] Include category tabs (all/images/news/etc) in header
**Describe the bug**
The image tab has a different layout (CSS?) from the other tabs like "all", "videos", "news". The text of the menu is bigger on the image tab.
Moreover, in the image tab I'm not able to access maps nor books, those links aren't displayed.

image
image

Is this intended? It's a bit disturbing to have a sudden change of layout like this.

**To Reproduce**
1. Do a search.
2. Click on the "image" link in the menu.
3. See different layout from the previous page

**Deployment Method**
- [ ] Heroku (one-click deploy)
- [ ] Docker
- [ ] `run` executable
- [x] pip/pipx
- [ ] Other: [describe setup]

**Version of Whoogle Search**
- [x] Latest build from [source] (i.e. GitHub, Docker Hub, pip, etc)
- [ ] Version [version number]
- [ ] Not sure

**Desktop (please complete the following information):**
 - OS: ArchLinux
 - Browser: Firefox
 - Version: 88.0 (64-bit)

**Additional context**

### PR DESCRIPTION
**Pull request by jacr13**
The rendering is similar in all the tabs. 

In addition, the tab disposition is the same across all the tabs.

If the translation is not present, the default names will be displayed (in English), some effort needs to be made here to add translation for all supported languages. It shouldn't be very hard to translate since they are very basic words.

This probably fixes #298. I intentionally removed the shopping tab because it requires some effort to render correctly (#136).

The header should be extended to add an extra row with different parameters to choice from (#322 and possibly #192)

### CODE REVIEW

File: app/utils/search.py
Comment (benbusby) [inline app/utils/search.py:125]: This doesn't appear to get used anywhere

```suggestion
```

File: app/utils/results.py
Comment (benbusby) [inline app/utils/results.py:252]: Can be simplified to

```suggestion
        query = full_query.replace(f'&tbm={search_type}', '')
```

File: app/templates/header.html
Comment (benbusby) [inline app/templates/header.html:71]: I assume this is a typo?

```suggestion
      <div class="desktop-tab-div">
```

File: app/static/css/header.css
Comment (benbusby) [inline app/static/css/header.css:110]: Typo here (mentioned below as well)

```suggestion
.desktop-tab-div {
```

File: app/filter.py
Comment (benbusby) [inline app/filter.py:291]: Can you elaborate what's going on here? I'm not sure I follow.

File: app/filter.py
Comment (benbusby) [inline app/filter.py:305]: Why is this necessary? It doesn't seem to affect the page from what I can tell, and things like `background-color` and `color` are being ignored altogether. I tried removing it and couldn't tell any difference.

File: app/templates/header.html
Comment (benbusby) [inline app/templates/header.html:88]: Is there a `mobile-tab-div` in the works?

File: app/static/settings/translations.json
Comment (benbusby) [inline app/static/settings/translations.json:43]: Equal translations are enforced in the tests, so these have to be translated to the existing languages. I'd like to automate this in the future, but haven't looked into it yet. 

That being said, I can take care of these. Whenever the feature is ready to merge, I'll go through and add the remaining translations before merging.

File: app/static/css/header.css
Comment (benbusby) [inline app/static/css/header.css:118]: Is this being applied? It's likely getting overwritten by the Whoogle theme. 

```suggestion
    background-color: #fff;
```

File: app/static/css/header.css
Comment (benbusby) [inline app/static/css/header.css:200]: From the class names, I'm guessing that these were extracted from the styles that Google includes with the tab bar. 

Would it be too much trouble to update the names of these classes to be more accurate? I'd like to avoid scenarios where we're using obfuscated class names provided by Google, since that ends up making the project harder to maintain if we need to tweak any of these later.

**Review by benbusby — CHANGES_REQUESTED** [review]
Awesome @jacr13, thanks for working on this feature. On desktop it seems to work really well. I did a quick review, the main things are:

- Are you working on a mobile tab replacement as well? Currently the mobile tab bar is removed but not replaced by anything
- Some styling elements are potentially not necessary, or need to be updated.

**Comment by jacr13** [comment]
Thanks @benbusby for reviewing the code!

This is a first version and of course it needs some rework. 
For the mobile version I have not done any testing, I will do when addressing the other issues!

I will not have too much time the next two weeks, so I may take a couple of weeks to fix yours remarks and implement the rest.

File: app/utils/search.py
Comment (jacr13) [reply app/utils/search.py:125]: It is used in the routes.py file when I call get_tab_contents [here](https://github.com/jacr13/whoogle-search/blob/7a86519b84bf3e232ae1268893e87e962e2b987a/app/routes.py#L263).

File: app/utils/results.py
Comment (jacr13) [reply app/utils/results.py:252]: Good point!

File: app/filter.py
Comment (jacr13) [reply app/filter.py:291]: The max-width returned to the images tab is 700 (for body tag) which is different on all other tabs (736) so we need to replace it. The html styling is also missing and is needed to have the exact same formatting.

File: app/filter.py
Comment (jacr13) [reply app/filter.py:305]: The colors and background colors can be removed, the others are needed to have a uniform styling across the tabs (It is mainly needed to fix the text formatting). It is only used in the images tab when the view image option is disabled!

File: app/templates/header.html
Comment (jacr13) [reply app/templates/header.html:88]: I will work on this !

File: app/static/settings/translations.json
Comment (jacr13) [reply app/static/settings/translations.json:43]: Yes, I saw that the test didn't pass due to the keys not being the same on all languages, but I have just filled the English to have the "reference" keys.

File: app/static/css/header.css
Comment (jacr13) [reply app/static/css/header.css:118]: Maybe not, I have just taken all the css related to the tabs, I need to do some cleaning.

File: app/static/css/header.css
Comment (jacr13) [reply app/static/css/header.css:200]: Exact!

You are right, will fix it.

**Comment by jacr13** [comment]
I have added support for the mobile version and fixed the issues you highlighted before.

The translation still needs to be done for all the languages (except English).

**Comment by benbusby** [comment]
> I have added support for the mobile version and fixed the issues you highlighted before.
> 
> The translation still needs to be done for all the languages (except English).

Awesome! Will review this week.

**Comment by benbusby** [comment]
Just FYI -- I'm holding off on reviewing this for a bit longer since Google is in the middle of changing how results are formatted, and I want to make sure their formatting changes don't break this feature. I'll be sure to review once it seems like the new result formatting changes they're making are finished/permanent.

**Comment by DUOLabs333** [comment]
Problem I've been having -- between switching tabs, for a brief second, it switches to some font (maybe Times New Roman), before going back to the default font.

**Pull request merged**
